tasks/carto.py

```diff
@@ -799,25 +835,97 @@ class OBSMeta(Task):
         'timespan': ['''
 CREATE TABLE {obs_meta} AS
 SELECT numer_timespan::TEXT timespan_id,
-       numer_timespan::TEXT timespan_name,
-       NULL::TEXT timespan_description,
+       numer_timespan_alias::TEXT timespan_alias,
+       numer_timespan_name::TEXT timespan_name,
+       numer_timespan_description::TEXT timespan_description,
+       numer_timespan_range::DATERANGE timespan_range,
+       numer_timespan_weight::NUMERIC timespan_weight,
        NULL::JSONB timespan_tags, --FIRST(timespan_tags)::JSONB timespan_tags,
-       NULL::NUMERIC timespan_weight,
        NULL::JSONB timespan_extra,
        NULL::TEXT timespan_type,
        NULL::TEXT timespan_aggregate,
-       ARRAY_REMOVE(ARRAY_AGG(DISTINCT numer_id)::TEXT[], NULL) numers,
-       ARRAY_REMOVE(ARRAY_AGG(DISTINCT denom_id)::TEXT[], NULL) denoms,
-       ARRAY_REMOVE(ARRAY_AGG(DISTINCT geom_id)::TEXT[], NULL) geoms,
+       NULL::TEXT[] numers,
+       NULL::TEXT[] denoms,
+       NULL::TEXT[] geoms,
        NULL::Geometry(Geometry, 4326) the_geom, -- ST_Union(DISTINCT ST_SetSRID(the_geom, 4326)) the_geom
        NULL::Integer timespan_version
 FROM observatory.obs_meta_next
-GROUP BY numer_timespan;
+WHERE numer_timespan IS NOT NULL
+GROUP BY numer_timespan, numer_timespan_alias, numer_timespan_name,
+         numer_timespan_description, numer_timespan_range, numer_timespan_weight;
         ''',
         '''
 ALTER TABLE {obs_meta} ADD PRIMARY KEY (timespan_id);
         ''',
         '''
+INSERT INTO {obs_meta}
+(timespan_id, timespan_alias, timespan_name, timespan_description, timespan_range, timespan_weight,
+timespan_tags, timespan_extra, timespan_type, timespan_aggregate, numers, denoms, geoms, the_geom, timespan_version)
+SELECT denom_timespan::TEXT timespan_id,
+       denom_timespan_alias::TEXT timespan_alias,
+       denom_timespan_name::TEXT timespan_name,
+       denom_timespan_description::TEXT timespan_description,
+       denom_timespan_range::DATERANGE timespan_range,
+       denom_timespan_weight::NUMERIC timespan_weight,
+       NULL::JSONB timespan_tags, --FIRST(timespan_tags)::JSONB timespan_tags,
+       NULL::JSONB timespan_extra,
+       NULL::TEXT timespan_type,
```

Comment:
Why aren't we storing the type?

Reply:
We're storing timespan for all the types not segregated by type so we cant set the type yet ( backward compabtility)